### PR TITLE
Add function to validate a trade offer url

### PR DESCRIPTION
**Pull request by Koponya**
The validateOfferUrl function request the url, and check the validity of it
Example:
offers.validateOfferUrl('https://steamcommunity.com/tradeoffer/new/?partner=50938317&token=qJVu_Ssk', function(err, isValid) {
  if(err || !isValid) {
    console.log('URL seems to be invalid');
  }
  console.log('URL seems to be valid');
});

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -71,6 +71,25 @@ SteamTradeOffers.prototype.getOfferUrl = function(callback) {
...
+  }, function(error, response, body) {
+    if (error || (response && response.statusCode !== 200)) {
+      return callback(error || new Error(response.statusCode));
+    }
+    if (!body) {
+      return callback(new Error('Invalid Response'));
+    }
+
+    if (body.indexOf("trade_partner_info_block") == -1) {
+      return callback(new Error('Invalid Response'));
```
Comment (andrewda) [inline index.js]: Although this is technically an invalid response, it's nowhere near as invalid as not having a `body` (and it's half-expected if we're checking the validity of the trade offer URL). You should probably `callback(null, false)` here, instead.

File: index.js [hunk 2]
```diff
@@ -71,6 +71,25 @@ SteamTradeOffers.prototype.getOfferUrl = function(callback) {
...
+    uri: offerUrl
+  }, function(error, response, body) {
+    if (error || (response && response.statusCode !== 200)) {
+      return callback(error || new Error(response.statusCode));
+    }
+    if (!body) {
+      return callback(new Error('Invalid Response'));
+    }
+
+    if (body.indexOf("trade_partner_info_block") == -1) {
```
Comment (andrewda) [inline index.js]: Single quotes and triple equals (`===`).

File: index.js [hunk 3]
```diff
@@ -71,6 +71,25 @@ SteamTradeOffers.prototype.getOfferUrl = function(callback) {
...
+  this._requestCommunity.get({
+    uri: offerUrl
+  }, function(error, response, body) {
+    if (error || (response && response.statusCode !== 200)) {
+      return callback(error || new Error(response.statusCode));
+    }
+    if (!body) {
+      return callback(new Error('Invalid Response'));
+    }
+
```
Comment (andrewda) [inline index.js]: Probably remove this new line to fit the style of this repo.

**Comment by Koponya** [comment]
You are right, so I did it, and write the new function into the readme :)

**Pull request closed without merging**